cloudfoundry/gorouter
# Configure HealthCheck User-Agent header

# What

Allow configuring the User-Agent header recognised as a healthcheck request. For example the Amazon ELBs send header `ELB-HealthChecker/1.0` in their healthchecks.
This is essential to take advantage of the drain feature of gorouter and be able to shutdown gracefully without interrupting any connection.
The original "HTTP-Monitor/1.1" is maintained as default.

# How to review
* Run gorouter and configure using:

```
drain_wait: 60
healthcheck_user_agent: test_header
```

* Configure a backend app as per https://github.com/cloudfoundry/gorouter#example
* `curl my_first_url.vcap.me:8081` (app) and `curl -A "test_header" "http://localhost:8081"` (healthcheck) should return 200
* Send signal USR1 to the process: `pkill -USR1 gorouter`. This starts draining process that will last 60s until gorouter stops.
* During this time:
  * `curl my_first_url.vcap.me:8081` should return 200
  * `curl -A "test_header" "http://localhost:8081"` should return 503 (to notify an upstream load balancer to stop sending new connections)

# Note

This should be followed by a PR on routing-release to allow using the new `healthcheck_user_agent` property.
